awx/api/serializers.py

```diff
@@ -4419,6 +4555,333 @@ def validate(self, attrs):
         return accepted
 
 
+class BulkJobNodeSerializer(serializers.Serializer):
+    # if we can find out the user, we can filter down the UnifiedJobTemplate objects
+    unified_job_template = serializers.IntegerField(
```

Comment:
Kind of strange that you use a `PrimaryKeyRelatedField` for inventory but an `IntegerField` here

Reply:
Oh my bad. corrected that.

Reply:
@AlanCoding Actually there is a reason we did that. If we do a PrimaryKeyRelatedField it increases the number of DB queries we make, which is why we take it as integer and later convert it into a object in get_objectified_jobs. That way we keep the number of queries less even with large number of jobs being launched via bulk job

Reply:
ah, to wrap this conversation up, the inventory field I referenced was on `BulkJobLaunchSerializer`, as opposed to this class which is a child serializer. The `PrimaryKeyRelatedField` have a history of performance problems, but if there is only 1 field for the serialized / view, a query or two would be acceptable, but the child serializer may repeat the query 1000 times in the request.